pinot-common/src/main/java/com/linkedin/pinot/common/data/FieldSpec.java

```diff
@@ -383,6 +383,9 @@ public int size() {
           return Float.BYTES;
         case DOUBLE:
           return Double.BYTES;
+        case BYTES:
```

Comment:
Don't add it here. This method should never be called for STRING and BYTES

Reply:
That requires this to be a derivedMetric, something we want to get rid of soon.